packages/runtime-core/src/apiCreateApp.ts

```diff
@@ -320,9 +322,22 @@ export function createAppAPI<HostElement>(
         }
       },
 
+      onUnmount(cleanupFn: () => void) {
+        if (typeof cleanupFn === 'function') pluginCleanupFns.push(cleanupFn)
+        else if (__DEV__) {
+          warn(
+            `Expected function as first argument to app.onUnmount(), but got ${typeof cleanupFn}`
+          )
+        }
+      },
       unmount() {
         if (isMounted) {
           render(null, app._container)
+          pluginCleanupFns.map(fn => {
```

Comment:
Let's just move this to after Vue's internal logic (line 345) so we don't need to worry about try...catching possible errors.

Also we can just do `for (const fn of pluginCleanupFns) fn()`.

Reply:
Done

Reply:
Did you forget to push the changes? 😅